homeassistant/components/roborock/image.py

```diff
@@ -108,6 +108,7 @@ def _create_image(self, map_bytes: bytes) -> bytes:
                 translation_domain=DOMAIN,
                 translation_key="map_failure",
             )
+        self.coordinator.maps[self.map_flag].current_room = parsed_map.vacuum_room
```

Comment:
I don't like the implementation of the new sensor, as it is dependent on the image entity.
The new sensor will never update if the user disables the map image entity.

Entities should not depend on each other to work.

Reply:
Unfortunately it has to - there is no where else that contains this information other than the map parsing. I'm not sure how I could avoid dependency. 

I guess I could potentially move the logic for getting the map and parsing it to the coordinator? And then the image would just have to grab from the coordinator? Would that be better?

Reply:
That would be a better solution indeed. The nice thing about coordinators, is that they stop working if no listeners (entities) are active as well.